internal/processing/passthrough_test.go

```diff
@@ -0,0 +1,68 @@
+package processing
+
+import (
+	"testing"
+
+	gatewayv2alpha1 "github.com/kyma-incubator/api-gateway/api/v2alpha1"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var (
+	apiName                 = "some-api"
+	apiUID        types.UID = "eab0f1c8-c417-11e9-bf11-4ac644044351"
+	apiNamespace            = "some-namespace"
+	apiAPIVersion           = "gateway.kyma-project.io/v2alpha1"
+	apiKind                 = "Gate"
+	apiGateway              = "some-gateway"
+	serviceName             = "example-service"
+	serviceHost             = "myService.myDomain.com"
+	servicePort   int32     = 8080
+)
+
+func TestGenerateVirtualService(t *testing.T) {
+	exampleAPI := &gatewayv2alpha1.Gate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      apiName,
+			UID:       apiUID,
+			Namespace: apiNamespace,
+		},
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: apiAPIVersion,
+			Kind:       apiKind,
+		},
+		Spec: gatewayv2alpha1.GateSpec{
+			Gateway: &apiGateway,
+			Service: &gatewayv2alpha1.Service{
+				Name: &serviceName,
+				Host: &serviceHost,
+				Port: &servicePort,
+			},
+		},
+	}
+	strategyPassthrough := &passthrough{}
+	vs := strategyPassthrough.generateVirtualService(exampleAPI)
+
```

Comment:
Hint: you can avoid passing `t` to all those assertions:
```suggestion
assert := assert.New(t)
```